korman/exporter/mesh.py

```diff
@@ -229,7 +229,14 @@ def _export_geometry(self, bo, mesh, materials, geospans):
                     source = mesh.vertices[vertex]
                     geoVertex = plGeometrySpan.TempVertex()
                     geoVertex.position = hsVector3(*source.co)
-                    geoVertex.normal = hsVector3(*source.normal)
+
+                    # If this face has smoothing, use the vertex normal
+                    # Otherwise, use the face normal
+                    if tessface.use_smooth:
```

Comment:
This is a really tight loop here that is already costly, performance wise. I would like to see assignment of `tessface.use_smooth` to a local value eg `smooth` so that we don't have to look up the `use_smooth` string in `tessface`'s dict every iteration.